src/main/scala/devices/debug/Debug.scala

```diff
@@ -1192,40 +1191,38 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
     // by harts executing code.
     //--------------------------------------------------------------
 
-    for (component <- 0 until nComponents) {
-      when (~io.dmactive || ~dmAuthenticated) {
-        haltedBitRegs(component) := false.B
-        resumeReqRegs(component) := false.B
+    when (~io.dmactive || ~dmAuthenticated) {
+        haltedBitRegs := 0.U
+        resumeReqRegs := 0.U
       }.otherwise {
-        // Hart Halt Notification Logic
-        when (hartIsInResetSync(component)) {
-          haltedBitRegs(component) := false.B
-          resumeReqRegs(component) := false.B
+
+        val hartHaltedIdIndex   = UIntToOH(hartSelFuncs.hartIdToHartSel(hartHaltedId))
+        val hartResumingIdIndex = UIntToOH(hartSelFuncs.hartIdToHartSel(hartResumingId))
+        val hartselIndex        = UIntToOH(io.innerCtrl.bits.hartsel)
+        when (hartIsInResetSync.reduce(_ | _)) {
+          haltedBitRegs := haltedBitRegs & ~(hartIsInResetSync.asUInt)
+          resumeReqRegs := resumeReqRegs & ~(hartIsInResetSync.asUInt)
         }.elsewhen (hartHaltedWrEn) {
```

Comment:
Same comment - this code is not functionally equivalent.  You must allow for some harts to be in reset while others are not.